Move bucketize_op to open source

olittle opened the pull request:
Differential Revision: D15145552

houseroad commented on caffe2/python/operator_test/bucketize_op_test.py:
```diff
@@ -0,0 +1,30 @@
+        op = core.CreateOperator('Bucketize',
+                                 ["X"], ["INDICES"],
+                                 boundaries=boundaries)
+        self.assertReferenceChecks(gc, op, [x, boundaries], ref)
```
Add a main function here, so we can run it separately.

houseroad reviewed (approved):
Looks good. Please check the inline comments.

olittle committed:
Move bucketize_op to open source

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/19952

Reviewed By: houseroad

Differential Revision: D15145552

fbshipit-source-id: d6181de239298727bb8007a8258d4526cc0513e3
